community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexProvider.java

```diff
@@ -231,9 +254,36 @@ public StoreMigrationParticipant storeMigrationParticipant( FileSystemAbstractio
         }
 
         @Override
-        public void drop( long indexId ) throws IOException
+        public void drop( long indexId, boolean archiveExistentIndex ) throws IOException
+        {
+            File rootIndexDirectory = directoryStructure.directoryForIndex( indexId );
+            if ( archiveExistentIndex && !FileUtils.isEmptyDirectory( rootIndexDirectory ) )
+            {
+                Map<String,String> env = MapUtil.stringMap( "create", "true" );
```

Comment:
Was this moved from lucene component? Would be nice to extract this archive thing as a utility and test it separately.

Reply:
ok, was thinking that it will be useful as separate utility as well, so will do that